src/decirest_handler_lib.erl

```diff
@@ -19,9 +22,35 @@
   add_default_allow_header/2
 ]).
 
+is_exported(Module, Function, ArityList) when is_list(ArityList) ->
+  lists:any(fun(R) -> R end,
+    [is_exported(Module, Function, Arity) || Arity <- ArityList ]
+  );
+
+is_exported(Module, Function, Arity) ->
+  erlang:function_exported(Module, Function, Arity).
+
+fetch_data(Req, State = #{module := Module}) ->
+  Bindings = cowboy_req:bindings(Req),
+  case is_exported(Module, fetch_data, 3) of
+    true ->
+      Module:fetch_data(Bindings, Req, State);
+    false ->
+      Module:fetch_data(Bindings, State)
+  end.
+
+delete_data(Req, State = #{module := Module}) ->
+  Bindings = cowboy_req:bindings(Req),
+  case is_exported(Module, delete_data, 3) of
+    true ->
+      Module:delete_data(Bindings, Req, State);
+    false ->
+      Module:delete_data(Bindings, State)
+  end.
+
```

Comment:
why are these moved into the lib is it to 
* reuse in collection later? generally delete isn't allowed in collection so it might be better to have int the single handler still
* keep same abstraction level in the handler callbacks? I like the idea, might be nice. But then we should add a todo to make that for the other parts as well, since there are patterns in the callbacks like case clauses on is_exported. If we remove that for delete/fetch we should delete it for other parts as well.
* Other?

Reply:
- you should be able to write your own handler and reuse the functions in the lib
- sure will add a TODO

Reply:
yea that's a nice aproach, so the handlers are just collections of libs, so it's quite easy to run your own.